src/databricks/labs/ucx/install.py

```diff
@@ -691,135 +692,102 @@ def install_on_account(self):
         # upload the json dump of workspace info in the .ucx folder
         ctx.account_workspaces.sync_workspace_info(installed_workspaces)
 
+    def get_workspace_contexts(self, other_workspace_id: int) -> list[WorkspaceContext]:
+        workspace_contexts = []
+        account_client = self._get_safe_account_client()
+        acct_ctx = AccountContext(account_client)
+        collection_workspace = account_client.workspaces.get(other_workspace_id)
+        if not acct_ctx.account_workspaces.can_administer(collection_workspace):
+            logger.error(f"User is not workspace admin of collection workspace {other_workspace_id}")
+            return []
+        installer = self._get_installer(collection_workspace)
+        if installer.config.installed_workspace_ids is None:
+            logger.error(f"No collection info found in the workspace {other_workspace_id}")
+            return []
+        for workspace_id in installer.config.installed_workspace_ids:
+            workspace = account_client.workspaces.get(workspace_id)
+            if not acct_ctx.account_workspaces.can_administer(workspace):
+                logger.error(f"User is not workspace admin of workspace {workspace_id}")
+                return []
+            workspace_client = account_client.get_workspace_client(workspace)
+            ctx = WorkspaceContext(workspace_client)
+            workspace_contexts.append(ctx)
+        return workspace_contexts
+
     @staticmethod
-    def _get_workspace(workspace_id: int, ids_to_workspace: dict[int, Workspace]) -> Workspace:
-        try:
-            workspace = ids_to_workspace[workspace_id]
-            return workspace
-        except KeyError:
-            msg = "Current workspace is not known, Please run as account-admin: databricks labs ucx sync-workspace-info"
-            raise KeyError(msg) from None
+    def _validate_collection(workspace_ids: list[int], ids_to_workspace: dict[int, Workspace]) -> bool:
+        # validate if the passed list of workspace ids, the user has workspace admin access on
+        # if not, then cant join a collection
+        for workspace_id in workspace_ids:
+            try:
+                ids_to_workspace[workspace_id]
+            except KeyError:
```

Comment:
```suggestion
            if workspace_id not in ids_to_workspace:
```
why do we need try-catch?..